src/search/sql_parser.h

```diff
@@ -84,7 +89,8 @@ struct Limit : string<'l', 'i', 'm', 'i', 't'> {};
 
 struct WhereClause : seq<Where, QueryExpr> {};
 struct AscOrDesc : sor<Asc, Desc> {};
-struct OrderByClause : seq<OrderBy, WSPad<Identifier>, opt<WSPad<AscOrDesc>>> {};
+struct OrderByExpr : sor<WSPad<VectorCompareExpr>, seq<WSPad<Identifier>, opt<WSPad<AscOrDesc>>>> {};
```

Comment:
Also here seems it's allowed to pass an ASC/DESC after `VectorCompareExpr`. We can either prevent it in sql::Transformer or change the syntax definition.

Reply:
This is functionally correct since `opt<WSPad<AscOrDesc>>` is wrapped into`seq<WSPad<Identifier>, [...]>` but it is unreadable, so made it to the following and add corresponding test
```
struct SortableFieldExpr : seq<WSPad<Identifier>, opt<AscOrDesc>> {};
struct OrderByExpr : sor<WSPad<VectorCompareExpr>, WSPad<SortableFieldExpr>> {};
```

Reply:
Ahh got it.